src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Utilities/DataFlowExtensions/DataFlowExtensionMethodCaller.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Utilities.DataFlowExtensions
+{
+    /// <summary>
+    /// <see cref="DataFlowExtensionMethodCaller"/> is a caller of a <seealso cref="IDataFlowExtensionWrapper"/>,
+    /// which wraps a call to extension methods defined in DataFlowExtensions. These methods cannot be mocked since
+    /// these are static methods. Hence we are covering the static methods with wrapper which makes the behavior non-static
+    /// which will then let us mock
+    /// </summary>
+    public class DataFlowExtensionMethodCaller
```

Comment:
It's not the same. Everywhere else, we push mocks into constructors and tests use the same constructors and plug-in as the product code.

Here you're injecting a mock into a property that is solely there for test code only. It's a pattern that I've seen elsewhere in Roslyn it makes it hard to reason about the code - as you've now got test only paths. Inject it in the constructor.